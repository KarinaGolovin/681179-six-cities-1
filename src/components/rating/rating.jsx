import React from 'react';
import PropTypes from 'prop-types';

export const Rating = ({rating = 0, classes = {}}) => {
  return (
    <div className={`rating ${classes.container || ``}`}>
      <div className={`rating__stars ${classes.stars || ``}`}>
        <span style={{
          width: `${ratingInPercents(rating)}%`
        }}>
        </span>
        <span className="visually-hidden">Rating</span>
      </div>
      {(classes.value ? <span className={`rating__value ${classes.value}`}>{rating}</span> : null)}
    </div>
  );
};

const ratingInPercents = (rating, max = 5) => {
  return Math.round(rating) / max * 100;
};

Rating.propTypes = {
  rating: PropTypes.number,
  classes: PropTypes.object,
  value: PropTypes.bool
};
