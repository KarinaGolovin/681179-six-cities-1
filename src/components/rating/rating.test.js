import React from 'react';
import renderer from 'react-test-renderer';
import {Rating} from './rating.jsx';

it(`Rating correctly renders`, () => {
  const component = renderer.create(
      <Rating rating={3.5} />
  ).toJSON();

  expect(component).toMatchSnapshot();
});
