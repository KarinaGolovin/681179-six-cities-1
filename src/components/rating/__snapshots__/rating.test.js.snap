// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating correctly renders 1`] = `
<div
  className="rating "
>
  <div
    className="rating__stars "
  >
    <span
      style={
        Object {
          "width": "80%",
        }
      }
    />
    <span
      className="visually-hidden"
    >
      Rating
    </span>
  </div>
</div>
`;
